data-client/src/main/java/org/wikipedia/gallery/ImageInfo.java

```diff
@@ -73,4 +90,25 @@ public int getHeight() {
     @Nullable public ExtMetadata getMetadata() {
         return metadata;
     }
+
+    /**
+     * Updates the ThumbUrl if image dimensions are not sufficient.
```

Comment:
Readers might not know what `not sufficient` means, so could you please say it more explicitly? Do not hesitate to explain the specific case of panoramic images.